### PR TITLE
Logistic Regression from scratch

### PR DESCRIPTION
**Pull request by Naman1233**
Logistic regression with SAG (stochastic average gradient descent) optimizer. User may require to give unbiased data (i.e. b = 0). Although fastest algorithm in ML. SAG does 1500 epochs within a milisecond time with acio.

### CODE REVIEW

File: model/scratch/dffml_model_scratch/logisticregression.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    predict: Feature = field("Label or the value to be predicted")
+    features: Features = field("Features to train on")
+    directory: pathlib.Path = field(
+        "Directory where state should be saved",
+        default=pathlib.Path("~", ".cache", "dffml", "scratch"),
+    )
+
+
+@entrypoint("scratchlgr")
+class LogisticRegression(SimpleModel):
```
Comment (pdxjohnny) [inline model/scratch/dffml_model_scratch/logisticregression.py]: There needs to be some explanation in the docstring of what makes this different than `SLR`.

File: model/scratch/dffml_model_scratch/logisticregression.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            gradJ = np.sum(x * f)
+            w = w - learning_rate * gradJ / len(x)
+        error = 0
+        for _id in range(len(x)):
+            yhat = x[_id] * w + b > 0.5
+            if yhat:
+                yhat = 1
+            else:
+                yhat = 0
+            if yhat != y[_id]:
```
[the remark this answers is not in the document]
Comment (Naman1233) [reply model/scratch/dffml_model_scratch/logisticregression.py]: ok _id -> x_id because id is an inbuilt function.
